### DISCUSSION TITLE
Bugs findings in makers platform

### DISCUSSION THREAD

**Discussion by marioekaputta** [bahasa-ai/otomatis-guidance#9] [Show and tell]
Here are a few bugs I found after using otomatis maker:
1. Wrong thumbnail. The app shows the telegram thumbnails instead of the google sheets. (Note: there's a telegram action in the scenario before, but I deleted it)
![1](https://user-images.githubusercontent.com/71392755/128161276-3211cc01-f8c8-4a59-ad0f-71ef17ff5556.png)

2. Gmail recipient address field shows an option from sheet data. This will lead to an error if the user places it towards the field.
![6](https://user-images.githubusercontent.com/71392755/128161627-fead6e13-04c3-4c3f-91e4-c1cdb9bd452b.png)

3. The warning state didn't change after I do the test dan retest on the data filter. However, the scenario still works well.
![4](https://user-images.githubusercontent.com/71392755/128161841-2a7a7918-77f4-4bc2-8627-1e613542f2b4.png)

4. Spam of action. I didn't trigger my scenario, but the action keeps spamming my email and telegram. (Scenario: when values change on google sheets -> send email -> send telegram messages). Earlier the action didn't work so I do some tests and reset the on/off switch. It works properly in the end, but not long after that, the spam happens.
![5](https://user-images.githubusercontent.com/71392755/128162491-c3d08f7b-5fce-4547-b0ed-1e5dd464f999.png)

5. Data numbers (with format) always return false. Since I used the google sheet as the trigger, I make a column with number value with format (e.g 12,000.00). I used data filter and conditional branching to compare the value but it always returns false. I also try to convert the param value with the format but it still returns false.
![3](https://user-images.githubusercontent.com/71392755/128163285-d9c3dd7e-0043-4270-8a50-561aa4c41d73.png)

6. Scenario with conditional branching has some delay action
7. Sometimes after making any changes to the action, we must run the test otherwise it will not work properly (works like it's not edited and the settings will be reset). The user must be notified about this.

**Comment by mgilangjanuar** [comment]
Wow, so many bugs in 1 thread 🤣

**Comment by nurulai** [comment]
> 1. Wrong thumbnail. The app shows the telegram thumbnails instead of the google sheets. (Note: there's a telegram action in the scenario before, but I deleted it)
>    ![1](https://user-images.githubusercontent.com/71392755/128161276-3211cc01-f8c8-4a59-ad0f-71ef17ff5556.png)

ini udh fix kayanya, gini kn ya mksudnya expectednya?

https://user-images.githubusercontent.com/62420151/129303890-ac96c226-ae1e-4d09-ade4-b4e508515945.mp4

**Reply by marioekaputta** [reply]
Sebenernya enggak kak. Di scenario aku tuh gaada trigger/action dari telegram justru, eh thumbnailnya malah telegram tapi. Sebelumnya sempet ada sih telegram as action, tp udah ku delete. Harusnya itu thumbnailnya google sheet

**Reply by nurulai** [reply]
i see,, nnti kalo nemu lg kabarin yaa,,

https://user-images.githubusercontent.com/62420151/129304362-d1c84caa-dcf5-42b1-97f6-eb8341f79458.mp4

udh tak coba ganti trigger, trus tumbnailnya udh expected

**Reply by marioekaputta** [reply]
Okayyy

**Comment by nurulai** [comment]
> 2. Gmail recipient address field shows an option from sheet data. This will lead to an error if the user places it towards the field.
>   ![6](https://user-images.githubusercontent.com/71392755/128161627-fead6e13-04c3-4c3f-91e4-c1cdb9bd452b.png)

ini mksudnya maunya ga ada optionsnya kah? cc ka @jundialwan 

krn yg skrg smua field pasti ada option dri hasil test state sblomnya

**Reply by marioekaputta** [reply]
Agree. Pas bikin thread ini aku belum paham fungsinya dibikin option di alamat email apa hehehe, so I think this is not a bugs

**Comment by nurulai** [comment]
> 6\. Scenario with conditional branching has some delay action

yg ini mksudnya gmn ya?

**Reply by marioekaputta** [reply]
Ohh kemarin pas action yg pake if else condition lalu send email, dia ngirimnya agak delay gitu kak. Tapi lately udah enggak, maybe my internet connection waktu itu.

**Reply by mgilangjanuar** [reply]
Mungkin karna masih di staging juga 🤣

**Comment by nurulai** [comment]
> 7\. Sometimes after making any changes to the action, we must run the test otherwise it will not work properly (works like it's not edited and the settings will be reset). The user must be notified about this.

ka @jundialwan , apa setiap berhasil update params kasih notify yg biasa d pojok kanan atas itu klo change updated?

**Reply by jundialwan** [reply]
Ini refer to red notif yg suruh Maker ngetest dulu sebelum lanjut?

**Reply by nurulai** [reply]
mm, bukan yg ini siih ka
![image](https://user-images.githubusercontent.com/62420151/130960516-233577e5-fc93-425a-b3d5-262632bad291.png)

tpi kaya yg ini.. cuma text nya bilang.. your change updated.. (?)
![image](https://user-images.githubusercontent.com/62420151/130960607-7f3e71dc-e6d8-4a7e-87f3-0122cfcd4b7d.png)
